datadog_checks_dev/datadog_checks/dev/tooling/commands/metadata.py

```diff
@@ -0,0 +1,138 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+import csv
+
+import click
+
+from .utils import (
+    CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success, echo_waiting, echo_warning
+)
+from ..constants import get_root
+
+from ..utils import load_manifest, get_valid_checks
+from ...utils import file_exists, dir_exists, resolve_path
+
+METADATA_FILE = 'metadata.csv'
+
+REQUIRED_HEADERS = {
+    'metric_name',
+    'metric_type',
+    'description',
+    'orientation',
+    'integration',
+    'short_name'
+}
+
+OPTIONAL_HEADERS = {
+    'interval',
+    'unit_name',
+    'per_unit_name',
+}
+
+ALL_HEADERS = REQUIRED_HEADERS | OPTIONAL_HEADERS
+
+
+@click.group(
+    context_settings=CONTEXT_SETTINGS,
+    short_help="Manage metadata files"
+)
+def metadata():
+    pass
+
+
+@metadata.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Validate `metadata.csv` files, takes optional `check` argument'
+)
+@click.argument('check', required=False)
+@click.pass_context
+def verify(ctx, check):
+    """Validates metadata.csv files
+
+    If `check` is specified, only the check will be validated, otherwise all matching files in directory.
+    """
+    check_list = []
+
+    root = get_root()
+
+    if check:
+        path = resolve_path(os.path.join(root, check))
+        if not dir_exists(path):
+            abort(
+                'Directory `{}` does not exist. Be sure to `ddev config set {repo} '
+                'path/to/integrations-{repo}`.'.format(path, repo=ctx.obj['repo_choice'])
+            )
+        else:
+            check_list = [check]
+    else:
+        check_list = [x for x in sorted(get_valid_checks())]
+
+    for current_check in check_list:
+        if current_check.startswith('datadog_checks_'):
+            continue
+
+        # get any manifest info needed for validation
+        manifest = load_manifest(current_check)
+        try:
+            metric_prefix = manifest['metric_prefix']
+        except KeyError:
+            echo_info('{}: metric_prefix does not exist in manifest'.format(current_check))
+            metric_prefix = None
+
+        path = resolve_path(os.path.join(root, current_check, METADATA_FILE))
+        if not file_exists(path):
+            echo_failure('Missing metadata file at: {}'.format(path))
+        else:
+            # To make logging less verbose, common errors are counted for current check
+            metric_prefix_count = dict()
+            empty_count = dict()
+            duplicate_set = set()
+
+            with open(path) as f:
+                reader = csv.DictReader(f, delimiter=',')
+
+                for row in reader:
+                    # duplicates
+                    if str(row) not in duplicate_set:
```

Comment:
instead of checking that the entire row is unique, I think we should just make sure that the `metric_name` is unique.